File: 66/66.js
var ans = [];

// This wont work since some of the min values of x are very large
// Also need to use a big int language/library
for(var D = 2; D <= 1000; D++) {
  if(Math.sqrt(D) % 1 === 0) {
    continue;
  }
  var y = 1;
  var x = Math.floor(Math.sqrt(D));
  var pell = x*x - D*(y*y);
  while(pell != 1) {
    if(pell < 1) {
      x++;
      y = 1;
    } else {
      y++;
    }
    pell = x*x - D*(y*y);
  }
  ans.push(x);
}

console.log(ans);

File: 66/66.py
def pell(d):
    p, k, x1, y, sd = 1, 1, 1, 0, d ** 0.5
    if sd == int(sd):
        return 0
    while k != 1 or y == 0:
        p = k * (p/k+1) - p
        p = p - int((p - sd)/k) * k
 
        x = (p*x1 + d*y) / abs(k)
        y = (p*y + x1) / abs(k)
        k = (p*p - d) / k
        x1 = x
    return x

ans = 1
for x in range(2, 1000):
    if pell(x) > ans:
        ans = pell(x)
        print x
